### ISSUE TITLE
inference 

### ISSUE THREAD

**Issue by superskrem** [joaopauloschuler/two-path-noise-lab-plant-disease#1]
how do you use the model for inference/test?

**Comment by joaopauloschuler** [comment]
You can load the model with cai.models.load_kereas_model.

Then, you can call **predict**:
https://www.activestate.com/resources/quick-reads/how-to-use-a-model-to-do-predictions-with-keras/

**Comment by superskrem** [comment]
> You can load the model with cai.models.load_kereas_model.
> 
> Then, you can call **predict**: https://www.activestate.com/resources/quick-reads/how-to-use-a-model-to-do-predictions-with-keras/

I did use this method but the prediction is always the same with deferent images  
always class 13 with confidence of 5.36  
even if I give it an image from the training data set  
and the validation and accuracy while training were 98%+  
so, i really don't know where is the problem

**Comment by superskrem** [comment]
would you please share with me the inference code ?

**Comment by superskrem** [comment]
this is how i do the prediction
```
imgs = cv2.imread("/content/0a334ae6-bea3-4453-b200-85e082794d56___GH_HL Leaf 310.1_flipTB.JPG")
imgs=cv2.resize(imgs, (224,224))
imm_array=np.array(imgs)
imm_array=imm_array/255
imm_array=imm_array.reshape(1, 224, 224, 3)
model = keras.models.load_model('/content/models/best.h5',custom_objects={'CopyChannels': cai.layers.CopyChannels})
predictions = model.predict(imm_array)
prediction_score = tf.nn.softmax(predictions)
predicated_class = np.argmax(prediction_score)
print(
    "This image most likely belongs to {} with a {:.2f} percent confidence."
    .format(classes[predicated_class], 100 * np.max(prediction_score)))
```
if you can help i will really appreciated

**Comment by joaopauloschuler** [comment]
@superskrem,
Thank you for sharing the code!

This line is not required (in the case that you are using any of my models):
`prediction_score = tf.nn.softmax(predictions)`

My models already contain a softmax layer.

Also, try to `print predictions` (the entire array) . You can follow this example: https://note.nkmk.me/en/python-numpy-ndarray-ndim-shape-size/

`predictions` might be a 2D array. Printing will give you clues.

**Comment by joaopauloschuler** [comment]
You can also try `print("prediction shape:", predictions.shape)`

**Comment by superskrem** [comment]
@joaopauloschuler,

thank you for the soft max hint now at least it's show a proper confidence value now.

and here is the prediction:
just keep in mind that the size is 49 because i added classes of my own.

```
prediction shape: (1, 49)
prediction dimensions: 2
prediction size: 49
prediction is: [[2.82565230e-22 1.34295552e-08 6.68176566e-04 1.00156047e-11
  4.87763302e-12 2.26236806e-20 8.16007457e-08 3.87023995e-18
  8.02653079e-16 7.38040271e-14 1.69374557e-18 7.32521545e-22
  7.68097777e-07 7.02550104e-17 8.92864943e-26 4.00462947e-13
  3.58292276e-08 9.37797511e-19 2.92795546e-13 3.18196776e-08
  2.78314326e-27 4.33391849e-25 2.17728791e-09 1.21087239e-17
  4.62941398e-06 5.62392364e-13 3.84908414e-17 1.27745914e-09
  2.93564414e-17 2.63131240e-18 9.09240238e-17 1.60283187e-15
  2.17595975e-16 1.89048736e-21 9.99283850e-01 7.87050564e-23
  6.14940220e-20 7.94256411e-22 9.80635520e-18 1.15108601e-15
  1.78264155e-13 6.94972925e-07 3.52296785e-17 2.15242401e-15
  4.17443916e-05 8.48204387e-23 6.23894465e-08 1.06473779e-17
  1.11528075e-20]]
prediction length: 49
```
i did flatten the prediction to be a 1D and tried that but sadly with same results

**Comment by superskrem** [comment]
and this is the output of this part

```
predicated_class = np.argmax(result)
print(predicated_class)
print(
    "This image most likely belongs to {} with a {:.2f} percent confidence."
    .format(classes[predicated_class], 100 * np.max(result)))
```


`34`
`This image most likely belongs to Strawberry___Leaf_scorch with a 99.93 percent confidence.`

**Comment by superskrem** [comment]
it work now.
all i had to do is convert the image from BGR to RGB then to LAB before doing the prediction

**Comment by joaopauloschuler** [comment]
AH! I didn't realize that you are using LAB (good choice).

@superskrem, thank you for sharing the outcome!
